File: src/net/eduard/api/lib/modules/Animation.java
package net.eduard.api.lib.modules;

import org.bukkit.entity.ArmorStand;
import org.bukkit.util.EulerAngle;

/**
 * Sistema de animar {@link ArmorStand}
 * 
 * @author Eduard
 * @version 1.0
 * @since 2.5
 */
public class Animation {

	private double one_degree = (Math.PI / 180);

	private ArmorStand stand;

	public ArmorStand getStand() {
		return stand;
	}

	public Animation(ArmorStand stand) {
		setStand(stand);
	}

	public void setStand(ArmorStand stand) {
		this.stand = stand;
	}

	public void moveHead(double x, double y, double z) {
		EulerAngle headAngule = stand.getHeadPose().add(x, y, z);
		stand.setHeadPose(headAngule);

	}

	public void moveHeadFront(int degrees) {
		moveHead(one_degree * degrees, 0, 0);
	}

	public void moveHeadBack(int degrees) {
		moveHead(-one_degree * degrees, 0, 0);
	}

	public void moveHeadUp(int degrees) {
		moveHead(0, 0, one_degree * degrees);
	}

	public void moveHeadDown(int degrees) {
		moveHead(0, 0, -one_degree * degrees);
	}

	public void moveHeadLeft(int degrees) {
		moveHead(0, one_degree * degrees, 0);
	}

	public void moveHeadRight(int degrees) {
		moveHead(0, -one_degree * degrees, 0);
	}

}

File: src/main/java/net/eduard/api/lib/game/Animation.kt
package net.eduard.api.lib.game

import org.bukkit.entity.ArmorStand

/**
 * Sistema de animar [ArmorStand]
 *
 * @author Eduard
 * @version 1.0
 * @since 2.5
 */
class Animation(val stand: ArmorStand) {

    private val one_degree = Math.PI / 180


    fun moveHead(x: Double, y: Double, z: Double) {
        val headAngule = stand.headPose.add(x, y, z)
        stand.headPose = headAngule
    }

    fun moveHeadFront(degrees: Int) {
        moveHead(one_degree * degrees, 0.0, 0.0)
    }

    fun moveHeadBack(degrees: Int) {
        moveHead(-one_degree * degrees, 0.0, 0.0)
    }

    fun moveHeadUp(degrees: Int) {
        moveHead(0.0, 0.0, one_degree * degrees)
    }

    fun moveHeadDown(degrees: Int) {
        moveHead(0.0, 0.0, -one_degree * degrees)
    }

    fun moveHeadLeft(degrees: Int) {
        moveHead(0.0, one_degree * degrees, 0.0)
    }

    fun moveHeadRight(degrees: Int) {
        moveHead(0.0, -one_degree * degrees, 0.0)
    }

}